In [1]:
import pandas as pd

In [11]:
RAG = r"C:\\Users\\kuzne\\Documents\\Python_repo\\2025_01_dissertation\\2025_dissertation\\data\\2025-06 hybrid search\\psychiatry_train_dataset_groq_mistral-saba-24b_answered_rag_answer_correctness_evaluated.csv"

In [12]:
VANILLA = r"C:\\Users\\kuzne\\Documents\\Python_repo\\2025_01_dissertation\\2025_dissertation\\data\\2025-06 hybrid search\\psychiatry_train_dataset_groq_mistral-saba-24b_answered_vanilla_answer_correctness_evaluated.csv"

In [13]:
# Read the files
rag_df = pd.read_csv(RAG)  # Replace with actual filename
vanilla_df = pd.read_csv(VANILLA)  # Replace with actual filename

# Calculate means
rag_mean = rag_df['Mean Answer Correctness for RAG'].mean()
vanilla_mean = vanilla_df['Mean Answer Correctness for vanilla'].mean()

print(f"RAG Mean Answer Correctness: {rag_mean}")
print(f"Vanilla Mean Answer Correctness: {vanilla_mean}")

RAG Mean Answer Correctness: 0.4384988387275732
Vanilla Mean Answer Correctness: 0.4564869899367323


The very first results were:  
RAG Mean Answer Correctness: 0.4123253574398246  
Vanilla Mean Answer Correctness: 0.4220597875819323  

Addition of Naive RAG deteriorates the answer quality. 


Hybrid Search pipeline:  
RAG Mean Answer Correctness: 0.4384988387275732  
Vanilla Mean Answer Correctness: 0.4564869899367323

Llama 3B

In [7]:
LLAMA_3B_VANILLA = r"C:\\Users\\kuzne\\Documents\\Python_repo\\2025_01_dissertation\\2025_dissertation\\data\\2025-06 hybrid search\\psychiatry_test_dataset_together_meta-llama_Llama-3.2-3B-Instruct-Turbo_answered_vanilla_answer_correctness_evaluated.csv"

LLAMA_3B_RAG = r"C:\\Users\\kuzne\\Documents\\Python_repo\\2025_01_dissertation\\2025_dissertation\\data\\2025-06 hybrid search\\psychiatry_test_dataset_together_meta-llama_Llama-3.2-3B-Instruct-Turbo_answered_rag_answer_correctness_evaluated.csv"


In [9]:
# Read the files
rag_df = pd.read_csv(LLAMA_3B_RAG)  # Replace with actual filename
vanilla_df = pd.read_csv(LLAMA_3B_VANILLA)  # Replace with actual filename

# Calculate means
llama3b_rag_mean = rag_df['Mean Answer Correctness for RAG'].mean()
llama3b_vanilla_mean = vanilla_df['Mean Answer Correctness for vanilla'].mean()

print(f"RAG Mean Answer Correctness: {llama3b_rag_mean}")
print(f"Vanilla Mean Answer Correctness: {llama3b_vanilla_mean}")

RAG Mean Answer Correctness: 0.40866393402516055
Vanilla Mean Answer Correctness: 0.4160153709655587


In [2]:
SIMPLE = r"C:\\Users\\kuzne\\Documents\\Python_repo\\2025_01_dissertation\\2025_dissertation\\data\\2025-06 hybrid search\\psychiatry_test_dataset_together_meta-llama_Llama-3.2-3B-Instruct-Turbo_answeredsimply_evaluated.csv"

In [3]:
simple = pd.read_csv(SIMPLE)  # Replace with actual filename

# Calculate means

simple_mean_vanilla = simple['GPT-based scoring for vanilla'].mean()
print(f"Simple Mean Answer Correctness: {simple_mean_vanilla}")

simple_mean_rag = simple['GPT-based scoring for RAG'].mean()
print(f"Simple Mean Answer Correctness for RAG: {simple_mean_rag}")




Simple Mean Answer Correctness: 0.5270880361173815
Simple Mean Answer Correctness for RAG: 0.4345372460496614
